### DISCUSSION TITLE
What would I have to change if packages aren't recognized?

### DISCUSSION THREAD

**Discussion by CanePlayz** [microsoft/winget-cli#3743] [General] [answered]
I have some packages installed which are also available from winget, but aren't picked up by it. `winget list` for those packages yields

```
Name                                          ID                                                   Version          Source
---------------------------------------------------------------------------------------------------------------------------
MyHeritage Family Tree Builder                Family Tree Builder                                  8.0.0.8636
Google Play Spiele (Beta)                     GooglePlayGames                                      23.9.1265.3
ROCCAT SWARM                                  InstallShield_{6FB8005D-F942-4BFB-AD75-5DA08E5454D9} 1.94.330
```

At least that's what I'm assuming because it doesn't change the ID to the winget one (which it does for all the ones with `winget` under `Source`).

What would I have to change in the manifest to make that work? Unfortunately, I don't know at all how the matching of local packages to winget manifests is handled by winget.

**Comment by denelon** [comment, marked as answer]
WinGet compares metadata in the registry against manifests in configured source(s).

Several factors go into what constitutes a valid "match".

It could be productCode / upgradeCode or any of the other values under "AppsAndFeatures" entries.

**Reply by CanePlayz** [reply]
Yup, my problem was solved by Trenly in an issue, thanks

**Reply by denelon** [reply]
Thanks @Trenly!
